File: src/archive/CSESJava/src/main/java/org/allan7yin/intro/MissingNumber.java
package org.allan7yin.intro;

import java.util.ArrayList;
import java.util.Scanner;

public class MissingNumber {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();

        ArrayList<Integer> nums = new ArrayList<>();
        boolean[] seen = new boolean[n + 1];

        for (int i = 0; i < n; i++) {
            int num = scanner.nextInt();
            nums.add(num);
            seen[num] = true;
        }

        for (int i = 1; i <= n; i++) {
            if (!seen[i]) {
                System.out.println(i);
            }
        }

        scanner.close();
    }
}

File: src/archive/CSES/Intro/missing_number.cpp
#include <iostream>
#include <unordered_set>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> nums;
    vector<bool> seen(n + 1, false);
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        nums.push_back(num);
        seen[num] = true;
    }

    for (int i = 1; i <= n; i++) {
        if (!seen[i]) {
            cout << i << endl;
        }
    }
}